api-docs/client-files.md

```diff
@@ -0,0 +1,262 @@
+# Client files
+
+This api is provided to upload and keep track of client files.
+
+## Get a single client file
+
+```http
+GET /api/clients/:file/:id/files
+```
+
+### Response
+
+#### Success
+
+```json
+{
+  "createdBy": {
+    "firstName": "Sean",
+    "lastName": "White",
+    "fullName": "Sean White",
+    "timestamp": "2019-11-06T06:00:00.000Z"
+  },
+  "lastUpdatedBy": {
+    "firstName": "Sean",
+    "lastName": "White",
+    "fullName": "Sean White",
+    "timestamp": "2019-11-06T06:00:00.000Z"
+  },
+  "fileName": "file-1",
+  "filePath": "file://host/path"
+}
+```
+
+#### Not Found
+
+If there is no client with the provided id, you will get a 404 HTTP response, with the following error
+
+```json
+{
+  "errors": ["Could not find client with id 2"]
+}
+```
+
+If there is no file with the provided id, you will get a 404 HTTP response, with the following error
+
+```json
+{
+  "errors": ["Could not find file with #123"]
+}
+```
+
+&nbsp;
+
+---
+
+&nbsp;
+
+## Get all client files
+
+```http
+GET /api/clients/:id/files
+```
+
+### Response
+
+#### Success
+
+The response for all files will look identical to get a single file but will return a array of files
+
+```json
+{
+  "files": [
+    {
+      "createdBy": {
+        "firstName": "Sean",
+        "lastName": "White",
+        "fullName": "Sean White",
+        "timestamp": "2019-11-06T06:00:00.000Z"
+      },
+      "lastUpdatedBy": {
+        "firstName": "Sean",
+        "lastName": "White",
+        "fullName": "Sean White",
+        "timestamp": "2019-11-06T06:00:00.000Z"
+      },
+      "fileName": "file-1",
+      "filePath": "file://host/path"
+    },
+    "..."
+  ]
+}
+```
+
+#### Not Found
+
+If there is no client with the provided id, you will get a 404 HTTP response, with the following error
+
+```json
+{
+  "errors": ["Could not find client with id 2"]
+}
+```
+
+&nbsp;
+
+---
+
+&nbsp;
+
+## Create a single client file
+
+```http
+POST /api/client/file
```

Comment:
RESTful resources (the full url) can and should be reused for listing, creating, and deleting files. We should be using the same url for all of the POST, GET, and DELETE operations.

```suggestion
POST /api/clients/:clientId/files
```